### PR TITLE
Adds steps to the run REST API

### PR DESCRIPTION
**Pull request by hakonhall**
Adds a "steps" field to the
/application/v4/tenant/{tenant}/application/{application}/instance/{instance}/job/{jobtype}/run/{number}
REST API with the following structure:

  "steps": {
    "deployTester": {
      "status": "succeeded",
      "startMillis": 1577972067092
    },
    "deployReal": {
      "status": "succeeded",
      "startMillis": 1577972067092
    },
    ...
  }

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/JobRunner.java [hunk 1]
```diff
@@ -34,18 +34,21 @@
...
     @TestOnly
-    public JobRunner(Controller controller, Duration duration, JobControl jobControl, ExecutorService executors, StepRunner runner) {
+    public JobRunner(Controller controller, Duration duration, JobControl jobControl, ExecutorService executors,
+                     StepRunner runner, Clock clock) {
         super(controller, duration, jobControl);
         this.jobs = controller.jobController();
         this.jobs.setRunner(this::advance);
         this.executors = executors;
         this.runner = runner;
+        this.clock = clock;
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/JobRunner.java]: All `Maintainer`s have access to the controller's clock through `controller().clock()` already.

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTester.java [hunk 2]
```diff
@@ -90,7 +87,7 @@ public DeploymentTester(ControllerTester controllerTester) {
         cloud = (MockTesterCloud) tester.controller().jobController().cloud();
         var jobControl = new JobControl(tester.controller().curator());
         runner = new JobRunner(tester.controller(), Duration.ofDays(1), jobControl,
-                               JobRunnerTest.inThreadExecutor(), new InternalStepRunner(tester.controller()));
+                               JobRunnerTest.inThreadExecutor(), new InternalStepRunner(tester.controller()), tester.clock());
```
Comment (jonmv) [inline controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTester.java]: And the tester's clock is the same as the controller's, in this case.

**Review by jonmv — APPROVED** [review]
I'd like to see the extra clock reference go, but otherwise looks good :)

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/JobRunner.java (hunk 1 above)
Comment (hakonhall) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/JobRunner.java]: Fixed

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTester.java (hunk 2 above)
Comment (hakonhall) [reply controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTester.java]: Fixed

**Review by jonmv — APPROVED** [review]
Very good! :)

**Pull request merged**